bukuserver/translations/__init__.py

```diff
@@ -0,0 +1,78 @@
+import os
+import re
+from babel.messages.frontend import CommandLineInterface as pybabel
+try:
+    from buku import __version__
+except ImportError:
+    __version__ = None
+
+DOMAIN = 'messages'
+DIR = os.path.dirname(__file__)
+BUKUSERVER = os.path.dirname(DIR)
+MAPPING = os.path.join(DIR, 'babel.cfg')
+TEMPLATE = os.path.join(DIR, 'messages.pot')
+CUSTOM = os.path.join(DIR, 'messages_custom.pot')
+
+_EOL, _STR_EOL, _G_STR_EOL = r'$\r?\n?', r'"[^\r\n]*"$\r?\n?', r'"([^\r\n]*)"$\r?\n?'
+STRINGS = {
+    r'\bPROJECT VERSION\b': __version__ or '???',
+    r'(?<=for )PROJECT\b': 'bukuserver',
+    r'(?<=as the )PROJECT(?= project)': 'buku',
+    r'\bORGANIZATION\b': 'buku',
+    f'^# FIRST AUTHOR <EMAIL@ADDRESS>, [0-9]+.{_EOL}': '',
+    f'^#, fuzzy{_EOL}': '',
+    r'(?<=^"POT-Creation-Date: ).*(?=\\n"$)': '2024-09-12 00:00+0000',  # avoid git updates of unchanged translations
+}
+OLD_BLANK = re.compile(f'^(?:#~ msgctxt {_STR_EOL})?#~ msgid {_STR_EOL}#~ msgstr ""{_EOL}(?:{_EOL})?', re.MULTILINE)
+OLD = re.compile(f'^(?:#~ msgctxt {_G_STR_EOL})?#~ msgid {_G_STR_EOL}#~ msgstr {_G_STR_EOL}(?:{_EOL})?', re.MULTILINE)
+
+def replace_obsolete(text):
+    '''Removes *blank* obsolete entries, and restores re-added *blank* values from old obsolete ones'''
+    text = re.sub(OLD_BLANK, '', text)
+    for obsolete in re.finditer(OLD, text):
+        _ctxt, _id, _str = obsolete.groups()
+        ctxt_re = ('' if _ctxt is None else f'msgctxt "{re.escape(_ctxt)}"{_EOL}')
+        if m := re.search(f'^{ctxt_re}msgid "{re.escape(_id)}"{_EOL}msgstr "()"{_EOL}', text, re.MULTILINE):
+            text = (text[:m.start(1)] + _str + text[m.end(1):]).replace(obsolete.group(0), '', 1)
+    return text
+
+def translations_generate():
+    '''Generates and patches the messages.pot template file'''
+    pybabel().run(['', 'extract', '--no-wrap', f'--mapping-file={MAPPING}',
+                   '--keywords=_ _l _p:1c,2 _lp:1c,2 lazy_gettext', f'--output-file={TEMPLATE}', BUKUSERVER])
```

Comment:
`pybabel()` hardcodes argv handling without stripping first parameter in CLI, so when invoking from the code an extra string at the beginning is needed to represent the (unused) program name